### PR TITLE
[Feature] Support auto import modules from registry.

### PR DESCRIPTION
**Pull request by RangiLyu**
## Motivation

The registry now supports auto-import modules from the given location.

register_all_modules before running is no longer needed. The modules will be lazy-imported during building.

- [x] This PR can be merged after https://github.com/open-mmlab/mmengine/pull/643. The MMEngine version should be updated.

resolve https://github.com/open-mmlab/mmyolo/issues/312

### CODE REVIEW

File: configs/timm_example/retinanet_timm-efficientnet-b1_fpn_1x_coco.py [hunk 1]
```diff
@@ -4,9 +4,6 @@
     '../_base_/schedules/schedule_1x.py', '../_base_/default_runtime.py'
 ]
 
-# please install mmcls>=0.20.0
-# import mmcls.models to trigger register_module in mmcls
-custom_imports = dict(imports=['mmcls.models'], allow_failed_imports=False)
```
Comment (HAOCHENYE) [inline configs/timm_example/retinanet_timm-efficientnet-b1_fpn_1x_coco.py:9]: Does it mean this PR can only be merged after MMCls support auto import?

File: demo/image_demo.py [hunk 2]
```diff
@@ -34,9 +33,6 @@ def parse_args():
 
 
 def main(args):
-    # register all modules in mmdet into the registries
-    register_all_modules()
-
```
Comment (HAOCHENYE) [inline demo/image_demo.py:39]: Do we need to call `init_default_scope` here?

File: configs/timm_example/retinanet_timm-efficientnet-b1_fpn_1x_coco.py (hunk 1 above)
Comment (RangiLyu) [reply configs/timm_example/retinanet_timm-efficientnet-b1_fpn_1x_coco.py:9]: No

File: demo/image_demo.py (hunk 2 above)
Comment (RangiLyu) [reply demo/image_demo.py:39]: The default_scope is initialized in inference api.

File: tools/analysis_tools/benchmark.py [hunk 3]
```diff
@@ -105,14 +105,13 @@ def dataset_benchmark(args, cfg, distributed, logger):
...
 def main():
-    register_all_modules()
-
     args = parse_args()
     cfg = Config.fromfile(args.config)
-
     if args.cfg_options is not None:
         cfg.merge_from_dict(args.cfg_options)
 
+    init_default_scope(cfg.get())
```
Comment (hhaAndroid) [inline tools/analysis_tools/benchmark.py:113]: It should be uniformly written as init_default_scope(cfg.get('default_scope', 'mmdet'))

File: configs/timm_example/retinanet_timm-efficientnet-b1_fpn_1x_coco.py (hunk 1 above)
Comment (hhaAndroid) [reply configs/timm_example/retinanet_timm-efficientnet-b1_fpn_1x_coco.py:9]: No?

Comment (RangiLyu) [reply configs/timm_example/retinanet_timm-efficientnet-b1_fpn_1x_coco.py:9]: Add some TODOs in the config. No need to wait for mmcls.

File: tools/deployment/test_torchserver.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ def align_ts_output(inputs, metainfo, device):
 
 
 def main(args):
-    register_all_modules()
+    init_default_scope('mmdet')
```
Comment (hhaAndroid) [inline tools/deployment/test_torchserver.py:58]: Does it need to be unified as `init_default_scope(cfg.get('default_scope', 'mmdet'))`?

File: tools/deployment/mmdet_handler.py [hunk 5]
```diff
@@ -5,12 +5,12 @@
...
 import numpy as np
 import torch
+from mmengine.registry import init_default_scope
 from ts.torch_handler.base_handler import BaseHandler
 
 from mmdet.apis import inference_detector, init_detector
-from mmdet.utils import register_all_modules
 
-register_all_modules(True)
+init_default_scope('mmdet')
```
Comment (hhaAndroid) [inline tools/deployment/mmdet_handler.py:13]: this is a bit strange. need?

File: mmdet/registry.py [hunk 6]
```diff
@@ -32,50 +32,89 @@
...
 # manage visualizer backend
-VISBACKENDS = Registry('vis_backend', parent=MMENGINE_VISBACKENDS)
+VISBACKENDS = Registry(
+    'vis_backend',
+    parent=MMENGINE_VISBACKENDS,
+    locations=['mmdet.visualization'])
 
 # manage logprocessor
-LOG_PROCESSORS = Registry('log_processor', parent=MMENGINE_LOG_PROCESSORS)
+LOG_PROCESSORS = Registry(
```
Comment (hhaAndroid) [inline mmdet/registry.py:117]: For those that have never been used in mmdet, can the locations just be written casually?

File: requirements/mminstall.txt [hunk 7]
```diff
@@ -1,2 +1,2 @@
 mmcv>=2.0.0rc1,<2.1.0
```
Comment (BIGWangYuDong) [inline requirements/mminstall.txt:1]: rc4?

**Review by hhaAndroid — APPROVED** [review]

**Review by BIGWangYuDong — APPROVED** [review]
LGTM

**Pull request merged**